### PR TITLE
Use state algos

### PR DESCRIPTION
**Pull request by mvpatel2000**
Use state algos, so engine changes to algos can adjust these variables as needed. Needed for agent

### CODE REVIEW

**Comment by mvpatel2000** [comment]
This is going to change following offline discussion

**Comment by mvpatel2000** [comment]
Oops closed wrong PR

File: composer/trainer/trainer.py [hunk 1]
```diff
@@ -973,8 +973,8 @@ def __init__(
         self.state.evaluators = evaluators
 
         # Some algorithms require specific settings
-        self._backwards_create_graph = any(map(lambda x: x.backwards_create_graph, ensure_tuple(algorithms)))
-        self._find_unused_parameters = any(map(lambda x: x.find_unused_parameters, ensure_tuple(algorithms)))
+        self._backwards_create_graph = any(map(lambda x: x.backwards_create_graph, ensure_tuple(self.state.algorithms)))
+        self._find_unused_parameters = any(map(lambda x: x.find_unused_parameters, ensure_tuple(self.state.algorithms)))
```
Comment (dskhudia) [inline composer/trainer/trainer.py:977]: ensure_tuple is not needed. self.state__init__ takes care of it.

**Review by dskhudia — APPROVED** [review]

File: composer/trainer/trainer.py (hunk 1 above)
Comment (mvpatel2000) [reply composer/trainer/trainer.py:977]: Good catch!

**Pull request merged**